src/Client/Adapter/Proxy.php

```diff
@@ -156,8 +156,8 @@ public function write($method, $uri, $httpVer = '1.1', $headers = [], $body = ''
         // Build request headers
         if ($this->negotiated) {
             $path = $uri->getPath();
-            if ($uri->getQuery()) {
-                $path .= '?' . $uri->getQuery();
+            if ($query = $uri->getQuery()) {
```

Comment:
I really don't like doing assignment in conditionals, as it's an easy vector for mistakes. When I merge, I'll assign prior to the conditional.